src/main/java/com/ss/training/lms/controller/BorrowerController.java

```diff
@@ -92,23 +93,18 @@
 	public ResponseEntity<BookLoan> returnBook(@RequestBody BookLoan loan) {
         if(loan.getBookId() == null || loan.getBranchId() == null || loan.getCardNo() == null)
             return new ResponseEntity<BookLoan>(loan, HttpStatus.BAD_REQUEST);
-        HttpStatus status = HttpStatus.OK;
-        try {
-            borrowerService.returnBook(loan);
-        } catch (ClassNotFoundException | SQLException e) {
-            status = HttpStatus.INTERNAL_SERVER_ERROR;
-        }
-        return new ResponseEntity<BookLoan>(loan, status);
+
+        boolean doesLoanExist = borrowerService.checkIfLoanExists(loan);
+        if(!doesLoanExist)
+            return new ResponseEntity<BookLoan>(loan, HttpStatus.NOT_FOUND);
+        borrowerService.returnBook(loan);
+        return new ResponseEntity<BookLoan>(loan, HttpStatus.OK);
 	}
 
 	@RequestMapping(path="/lms/borrower/checkOutBook/{bookId}/branch/{branchId}/borrower/{cardNo}", method = RequestMethod.POST)
 	public HttpStatus checkOutBook(@PathVariable int bookId, @PathVariable int branchId, @PathVariable int cardNo) {
         HttpStatus status = HttpStatus.OK;
-        try {
-            borrowerService.checkOutBook(bookId, branchId, cardNo);
-        } catch (ClassNotFoundException | SQLException e) {
-            status = HttpStatus.INTERNAL_SERVER_ERROR;
-        }
+        borrowerService.checkOutBook(bookId, branchId, cardNo);
         return status;
```

Comment:
Shouldn't it be possible to return a status other than OK, for example if a path parameter is invalid or the selected branch has no copies of the selected book?

Reply:
I added a check for the no copies. I was trying to think of what it would mean to have an invalid request. 

if you put in strings it returns the json below which seems fine. I think any integer should be allowed as long as the check to see if the book is available exists. Let me know if i am missing something!
{
    "timestamp": "2020-05-13T18:42:49.749+0000",
    "status": 400,
    "error": "Bad Request",
    "message": "Failed to convert value of type 'java.lang.String' to required type 'int'; nested exception is java.lang.NumberFormatException: For input string: \"tre\"",
    "path": "/lms/borrower/checkOutBook/tre/branch/trg/borrower/h5"
}

Reply:
If the path parameters correspond to a book/branch/borrower that don't exist, the request would be invalid. I think the checkIfBookAvailable method takes care of the book and branch, but not the borrower.